php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1702 @@
+  }
+
+  public ExitTree completeExitExpression(InternalSyntaxToken exitOrDie, Optional<ExitTreeImpl> partial) {
+    return partial.isPresent() ? partial.get().complete(exitOrDie) : new ExitTreeImpl(exitOrDie);
+  }
+
+  public ExpressionTree combinedScalarOffset(ArrayInitializerTree arrayInitialiser, Optional<List<ArrayAccessTree>> offsets) {
+    ExpressionTree result = arrayInitialiser;
+    for (ArrayAccessTree offset : optionalList(offsets)) {
+      result = ((ArrayAccessTreeImpl) offset).complete(result);
+    }
+
+    return result;
+  }
+
+  public ExpressionTree postfixExpression(ExpressionTree expression, Optional<Object> optional) {
+    if (optional.isPresent()) {
+
+      if (optional.get() instanceof SyntaxToken) {
+        SyntaxToken operator = ((SyntaxToken) optional.get());
+
+        return new PostfixExpressionTreeImpl(
+          operator.text().equals(PHPPunctuator.INC.getValue()) ? Kind.POSTFIX_INCREMENT : Kind.POSTFIX_DECREMENT,
+          expression,
+          operator);
+
+      } else {
+        Tuple<InternalSyntaxToken, ExpressionTree> tuple = ((Tuple) optional.get());
+        return new BinaryExpressionTreeImpl(Kind.INSTANCE_OF, expression, tuple.first(), tuple.second);
+      }
+    }
+
+    return expression;
+  }
+
+  public AssignmentExpressionTree assignmentExpression(ExpressionTree lhs, InternalSyntaxToken operatorToken, ExpressionTree rhs) {
+    String operator = operatorToken.text();
+    Kind kind = Kind.ASSIGNMENT;
+
+    if ("*=".equals(operator)) {
+      kind = Kind.MULTIPLY_ASSIGNMENT;
+    } else if ("/=".equals(operator)) {
+      kind = Kind.DIVIDE_ASSIGNMENT;
+    } else if ("%=".equals(operator)) {
+      kind = Kind.REMAINDER_ASSIGNMENT;
+    } else if ("+=".equals(operator)) {
+      kind = Kind.PLUS_ASSIGNMENT;
+    } else if ("-=".equals(operator)) {
+      kind = Kind.MINUS_ASSIGNMENT;
+    } else if ("<<=".equals(operator)) {
+      kind = Kind.LEFT_SHIFT_ASSIGNMENT;
+    } else if (">>=".equals(operator)) {
+      kind = Kind.RIGHT_SHIFT_ASSIGNMENT;
+    } else if ("&=".equals(operator)) {
+      kind = Kind.AND_ASSIGNMENT;
+    } else if ("^=".equals(operator)) {
+      kind = Kind.XOR_ASSIGNMENT;
+    } else if ("|=".equals(operator)) {
+      kind = Kind.OR_ASSIGNMENT;
+    } else if (".=".equals(operator)) {
+      kind = Kind.CONCATENATION_ASSIGNMENT;
+    }
+
+    return new AssignmentExpressionTreeImpl(kind, lhs, operatorToken, rhs);
+  }
+
+  public AssignmentByReferenceTree assignmentByReference(ExpressionTree lhs, InternalSyntaxToken equToken, InternalSyntaxToken ampersandToken, ExpressionTree rhs) {
+    return new AssignmentByReferenceTreeImpl(lhs, equToken, ampersandToken, rhs);
+  }
+
+  public ConditionalExpressionTreeImpl newConditionalExpr(InternalSyntaxToken queryToken, Optional<ExpressionTree> trueExpression, InternalSyntaxToken colonToken, ExpressionTree falseExpression) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 196 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00103)